### ISSUE TITLE
Set org-mode default only for user-demanded buffers

### ISSUE THREAD

**Issue by bvraghav** [bvraghav/dotemacs#10]
## TLDR; ##
+ Find a way to identify whether a buffer creation request was user demanded or an automatic one; and
+ Trigger org-mode as default only in such a case!

# Details #
Org-mode is too overloaded to be a default major mode! Adapted from [`setup/org-setup.el`](https://github.com/bvraghav/dotemacs/blob/master/setup/org-setup.el#L125-L126)
```elisp
  ;; Define org mode as default
  (setq-default major-mode 'org-mode)
```
This piece of setup code breaks when *visiting* an org-mode (file) buffer containing a jupyter-based source code. That's because jupyter would try to find a kernelspec, from a shell command output captured into an emacs buffer. But if `org-mode` is default, it triggers an infinite-loop.
